### PR TITLE
added battery cover in build123d

### PR DESCRIPTION
**Pull request by przymi**

### CODE REVIEW

**Comment by jdegenstein** [comment]
Took a look and here is my version with a few suggestions, feel free to use any part or all parts as you see fit:
```py
from build123d import (
    BuildPart,
    BuildSketch,
    BuildLine,
    Box,
    Plane,
    Polyline,
    Location,
    Locations,
    Axis,
    Rot,
    mirror,
    make_face,
    extrude,
    fillet,
    chamfer,
)

COVER_WIDTH = 62
COVER_DEPTH = 37
COVER_HEIGHT = 2

POLY_WIDTH = 25
FILLET_SIZE = 2

BOX_WIDTH = 6

USE_TABS = True  ## False

with BuildPart() as cover:
    with BuildSketch() as s:
        Rectangle(COVER_WIDTH, COVER_DEPTH)
    extrude(amount=COVER_HEIGHT)  # position control, box is now NOT half below XY
    split(bisect_by=Plane.YZ)  # keep only half

    with BuildPart(mode=Mode.PRIVATE) as box_p: # construct private box on the origin and transform later
        with BuildSketch() as s:
            Rectangle(BOX_WIDTH, 6)
        extrude(amount=8)

        if USE_TABS: #placing tabs is easier because the selector is simpler and more robust
            fc2 = faces().filter_by(Axis.X)[-1] #get face center of small edge box
            plane = Plane(fc2).rotated((0, -5, 0)) #rotate -5 degrees about local y-axis
            with BuildSketch(plane) as tab_sk2:
                Rectangle(4, 2, align=(Align.CENTER, Align.MAX)) #align top edge to y-axis
            extrude(until=Until.PREVIOUS) # used extrude-until so that dimensions are not needed
        mirror(about=Plane.YZ) #leverage local centerline symmetry to mirror the tab

    loc = Location(
        (COVER_WIDTH / 2 - 8 - BOX_WIDTH / 2, -COVER_DEPTH / 2 - BOX_WIDTH / 2, 0)
    )
    with Locations(loc):
        add(box_p.part) #move the box into position

    plane = Plane.XZ.offset(-COVER_DEPTH / 2)
    with BuildSketch(plane) as poly_sk:
        with BuildLine(Location((COVER_WIDTH / 2 - POLY_WIDTH, COVER_HEIGHT))) as line:
            Polyline((0, 0), (POLY_WIDTH, 0), (20, 5), (5, 5), (0, 0))
        make_face()
    extrude(amount=COVER_HEIGHT)

    edgs = cover.part.edges().filter_by(Axis.X).group_by(Axis.Y)[2][0]
    fillet(edgs, FILLET_SIZE)
    edgs = cover.part.edges().filter_by(Axis.X).group_by(Axis.Y)[1][1]
    fillet(edgs, FILLET_SIZE)
    #eliminated one fillet with later mirror operation

    if USE_TABS:
        fc = poly_sk.sketch.face().center() # get trapezoid face center
        plane = Plane(
            origin=(fc.X, fc.Y, COVER_HEIGHT + 2), x_dir=(-1, 0, 0), z_dir=(0, 1, 0)
        ) # manually construct a plane with some of the face center coordinates
        with BuildSketch(plane.rotated((5, 0, 0))) as tab_sk: # create sketch on plane rotated about local y-axis
            Rectangle(4, 2, align=(Align.CENTER, Align.MAX))
        extrude(until=Until.PREVIOUS)

    mirror(about=Plane.YZ) #leverage CL symmetry to reduce code-duplication

if __name__ in ['__main__', 'temp']:
    show_object(cover)
```
The only caveat here is that extrude-until can be sometimes a little less stable than providing a dimension manually. In this case it seems to work very well, but just be advised. One could easily substitute a dimension manually in this case too.

File: build123d/cover.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+from build123d import (
+    BuildPart, BuildSketch, BuildLine, Box, Plane, Polyline, Location, Locations,
+    Axis, Rot,
+    mirror, make_face, extrude, fillet, chamfer
+    )
```
Comment (marcos-diaz) [inline build123d/cover.py]: Bracket either goes into previous line end, or this line without any indent.

File: build123d/cover.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+from ocp_vscode import show_object
```
Comment (marcos-diaz) [inline build123d/cover.py]: This possibly needs a try-catch to avoid failing in non VS enviroments?

File: build123d/cover.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+FILLET_SIZE = 2
+
+USE_TABS = True ## False
+
+with BuildPart() as cover:
+    Box(COVER_WIDTH, COVER_DEPTH, COVER_HEIGHT)
+
+    BOX_WIDTH = 6
+    with Locations(
+        (-COVER_WIDTH / 2 + 8 + BOX_WIDTH / 2, -COVER_DEPTH / 2 - BOX_WIDTH / 2, 3),
```
Comment (marcos-diaz) [inline build123d/cover.py]: Move these calculations to the constant sections and assign them a helpful constant name.

File: build123d/cover.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        (COVER_WIDTH / 2 - 8 - BOX_WIDTH / 2, -COVER_DEPTH / 2 - BOX_WIDTH / 2, 3)):
+        Box(BOX_WIDTH, 6, 8)
+
+    plane = Plane.XZ.offset(-COVER_DEPTH / 2)
```
Comment (marcos-diaz) [inline build123d/cover.py]: What this block does? All blocks need comments of intention

File: scripts/export_b123d.py [hunk 5]
```diff
@@ -21,3 +22,6 @@ def export(obj, filename):
 # Trigger L1/R1.
 export(trigger_r1.part, 'primary_015mm_trigger_R1')
 export(trigger_r1.part.mirror(Plane.YZ), 'primary_015mm_trigger_L1')
+
+# Battery Cover.
+export(cover.part, 'secondary_015mm_cover')
```
Comment (marcos-diaz) [inline scripts/export_b123d.py:27]: This is missing disabling the corresponding Blender export in other file

File: build123d/cover.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+TAB_ANGLE = 5 # degrees
+
+USE_TABS = True # only for friction hold of 3d-printed parts
+
+FILLET_SIZE = 2
+
+
+with BuildPart() as cover:
+    # Bottom box
+    with BuildSketch() as bottom_sk:
```
Comment (marcos-diaz) [inline build123d/cover.py]: Do not create `bottom_sk` variable if it won't be used

File: build123d/cover.py [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+        Rectangle(COVER_BOTTOM_WIDTH, COVER_BOTTOM_DEPTH)
+    extrude(amount=COVER_BOTTOM_HEIGHT)
+    split(bisect_by=Plane.YZ)  # keep only half, since everything will be mirrored in the end
+
+    # Front trapezoid
+    plane = Plane.XZ.offset(-COVER_BOTTOM_DEPTH / 2)
+    with BuildSketch(plane) as poly_sk:
```
Comment (marcos-diaz) [inline build123d/cover.py:46]: Same, unused variable

File: build123d/cover.py [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+            Rectangle(BOX_WIDTH, BOX_DEPTH)
+        extrude(amount=BOX_HEIGHT)
+
+        if USE_TABS:
+            box_face = faces().filter_by(Axis.X)[-1]
+            plane = Plane(box_face).rotated((0, -TAB_ANGLE, 0)) # rotate about local y-axis
+            with BuildSketch(plane):
+                Rectangle(TAB_WIDTH, TAB_HEIGHT, align=(Align.CENTER, Align.MAX)) # align top edge to y-axis
+            extrude(until=Until.PREVIOUS)
+        mirror(about=Plane.YZ) # leverage local centerline symmetry to mirror the tab
```
Comment (marcos-diaz) [inline build123d/cover.py:74]: Why is this mirrored manually instead of later in the final mirror step?

Comment (przymi) [reply build123d/cover.py:74]: This is mirroring the tab on the box. Lateron the box, including the tabs are mirrored in the final step

File: scripts/export_blender.py [hunk 9]
```diff
@@ -130,7 +130,6 @@ def export(self):
     Entry('DHat',             'primary_015mm_dhat'),
     Entry('Select',           'primary_007mm_select_4x', split=True),
     Entry('Case back',        'secondary_015mm_back', '-Z'),
-    Entry('Case cover',       'secondary_015mm_cover'),
```
Comment (marcos-diaz) [inline scripts/export_blender.py:133]: Remove also from `Makefile` line:
```
$(BLENDER) blender/case_cover.blend --background --python scripts/export_blender.py
```

File: build123d/cover.py (hunk 7 above)
Comment (przymi) [reply build123d/cover.py:46]: poly_sk is actually used further down, to determine the location of the tabs.

**Review by marcos-diaz — APPROVED** [review]

**Pull request merged**